### PR TITLE
Better submission status viewing

### PR DESCRIPTION
**Pull request by vishwakftw**
Also, I think the current way of displaying test case statuses aren’t very nice to look at.

- [x] read error messages from a file
- [x] use modals for status viewing
- [x] use badges for mini status

### CODE REVIEW

**Comment by prateekkumarweb** [comment]
Some screenshots:
image
image
image
image

**Comment by vishwakftw** [comment]
I've made some minor changes, pushing them

**Comment by vishwakftw** [comment]
Also, is it possible to line up a bunch of them (e.g., 3) in a line? This saves space.

File: judge/templates/judge/submission_detail.html [hunk 1]
```diff
@@ -145,17 +147,17 @@
                                         <div class="card shadow border-0">
                                             <div class="card-body px-lg-5 py-lg-5">
                                                 <div class="text-center mb-4">
-                                                    Test case
+                                                    Test Case {{ forloop.counter }}
```
Comment (prateekkumarweb) [inline judge/templates/judge/submission_detail.html]: Why case starts with capital letter?

Comment (vishwakftw) [reply judge/templates/judge/submission_detail.html]: Well, I thought it'd look better.

"Test Case" looks better than "Test case" :P

Comment (prateekkumarweb) [reply judge/templates/judge/submission_detail.html]: For me, Test case makes more sense. Anyways your call.

Comment (vishwakftw) [reply judge/templates/judge/submission_detail.html]: Let it be "Test Case".

**Comment by prateekkumarweb** [comment]
> Also, is it possible to line up a bunch of them (e.g., 3) in a line? This saves space.

It is possible. Currently, I am using tables for aligning. To make multiple, we won't' be able to use tables, and we will have to use columns and rows for responsive design. There are two possibilities here:

1. Fixed number of cols for each device configuration like 1 col for mobile and 2 cols for tab and 3 for desktops.
2. We let bootstrap decide the number of cols.

What do you want to have?

BTW you won't save paper by saving space. :stuck_out_tongue:

**Comment by vishwakftw** [comment]
I'll try out a few submissions and see if the message comes out correctly.

**Comment by vishwakftw** [comment]
> Fixed number of cols for each device configuration like 1 col for mobile and 2 cols for tab and 3 for desktops.

This sounds reasonable, but manual effort.

> We let bootstrap decide the number of cols.

Is this device agnostic or not?

**Comment by prateekkumarweb** [comment]
> This sounds reasonable, but manual effort.
Not at all.

> Is this device agnostic or not?
Depending on window size it will try to fit the number of cols.

**Comment by vishwakftw** [comment]
So, the messages are getting printed out correctly with proper indentation and all. I guess my work in this PR is done.

**Comment by vishwakftw** [comment]
@prateekkumarweb did you have time to wrap this PR up?

**Comment by prateekkumarweb** [comment]
I will be doing it tonight.


---- On Thu, 06 Jun 2019 11:32:12 +0530 notifications@github.com wrote ----



@prateekkumarweb did you have time to wrap this PR up?

**Comment by vishwakftw** [comment]
coolio

**Comment by vishwakftw** [comment]
I’ll split out my bits from this PR.

**Comment by prateekkumarweb** [comment]
> I’ll split out my bits from this PR.

What?

**Comment by vishwakftw** [comment]
Nvm

**Comment by prateekkumarweb** [comment]
image
image

**Comment by vishwakftw** [comment]
Wow beautiful.

**Comment by vishwakftw** [comment]
Are these the final changes?

**Comment by prateekkumarweb** [comment]
> Are these the final changes?

You say. Any other changes required?

**Comment by vishwakftw** [comment]
Nope, none come to mind. I’ll merge now.

**Pull request merged**